python/pyspark/sql/session.py

```diff
@@ -510,6 +578,12 @@ def createDataFrame(self, data, schema=None, samplingRatio=None, verifySchema=Tr
         except Exception:
             has_pandas = False
         if has_pandas and isinstance(data, pandas.DataFrame):
+            if self.conf.get("spark.sql.execution.arrow.enabled", "false").lower() == "true" \
+                    and len(data) > 0:
+                df = self._createFromPandasWithArrow(data, schema)
+                # Fallback to create DataFrame without arrow if return None
+                if df is not None:
```

Comment:
Shall we show some log message to users in this case?

Reply:
the PR from @ueshin better captures errors to show a warning, i'll merge that now